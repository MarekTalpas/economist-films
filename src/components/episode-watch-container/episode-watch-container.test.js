// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import EpisodeWatchContainer from './episode-watch-container';

test('EpisodeWatchContainer renders correctly', () => {
  const tree : string = renderer.create(<EpisodeWatchContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Checking fetch function: ', () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(() => {
      const p = new Promise((resolve) => {
        resolve({
          ok: true,
          status: 200,
          json: () => ({
            url: 'http',
          }),
        });
      });
      return p;
    });
  });
});