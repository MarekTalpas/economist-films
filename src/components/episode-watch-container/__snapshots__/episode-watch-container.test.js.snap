exports[`test EpisodeWatchContainer renders correctly 1`] = `
<div
  className="episode-watch">
  <div
    className="episode-watch__video"
    style={
      Object {
        "background": "#000 url(\"\") center/cover no-repeat",
      }
    } />
  <button
    className="episode-watch__watchnow-button">
    Watch now
  </button>
</div>
`;
